### PR TITLE
More consistently use directory index files

### PR DESCRIPTION
**Pull request by oprypin**
Another completely optional change, but I thought it was nice for file organization.

### CODE REVIEW

File: docs/syntax_and_semantics/annotations/built_in_annotations.md [hunk 1]
```diff
@@ -1,16 +1,5 @@
...
-* [Extern](#extern)
-* [ThreadLocal](#threadlocal)
-* [Packed](#packed)
-* [AlwaysInline](#alwaysinline)
-* [NoInline](#noinline)
-* [ReturnsTwice](#returnstwice)
-* [Raises](#raises)
-* [CallConvention](#callconvention)
-* [Flags](#flags)
-
```
Comment (oprypin) [inline docs/syntax_and_semantics/annotations/built_in_annotations.md:13]: Note: this manual list is obsoleted by the automatic right sidebar

Preview: https://oprypin.github.io/crystal-book/syntax_and_semantics/annotations/built_in_annotations.html

**Review by straight-shoota — APPROVED** [review]

**Comment by bcardiff** [comment]
Is mkdocs-redirects SEO friendly? Maybe instead of adding that we would upload the bucket configuration with redirect rules in this repo and update them on deploy.

When usiing dir/README.md I'm guessing ./dir/README is the link. If instead dir/README.md, an dir/index.md file is used, wouldn't that generate a ./dir link directly and would be nicer?

**Comment by oprypin** [comment]
> Is mkdocs-redirects SEO friendly? Maybe instead of adding that we would upload the bucket configuration with redirect rules in this repo and update them on deploy.

Seems reasonably friendly. I think the hassle of manual reconfiguration is not worth it.

`cat site/syntax_and_semantics/literals.html`
```html

<!doctype html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Redirecting...</title>
    <link rel="canonical" href="/reference/syntax_and_semantics/literals/index.html">
    <meta name="robots" content="noindex">
    <script>var anchor=window.location.hash.substr(1);location.href="/reference/syntax_and_semantics/literals/index.html"+(anchor?"#"+anchor:"")</script>
    <meta http-equiv="refresh" content="0; url=/reference/syntax_and_semantics/literals/index.html">
</head>
<body>
Redirecting...
</body>
</html>
```

> When usiing dir/README.md I'm guessing ./dir/README is the link. If instead dir/README.md, an dir/index.md file is used, wouldn't that generate a ./dir link directly and would be nicer?

For MkDocs, `README.md` 100% behaves like `index.md` would *only except* if `index.md` also exists. So no, it wouldn't change this.

> I'm guessing ./dir/README is the link

Well no, it's `/dir/index.html`.

> generate a ./dir link directly

MkDocs itself uses `index.html` even for normal links, unfortunately, so this small improvement would only break that convention.

**Comment by bcardiff** [comment]
Ok, cool. Thanks for clarifying.

**Pull request merged**